aerogear/unifiedpush-operator
# Create Push apps and variants in UPS based on equivalent CRs

JIRA: https://issues.jboss.org/browse/AEROGEAR-9022

----

## Verification steps

1. Deploy the operator:

```
make cluster/prepare
kubectl apply -n unifiedpush -f ./deploy/operator.yaml
```

2. Provision a UPS by creating a CR:

```
kubectl apply -n unifiedpush -f ./deploy/crds/push_v1alpha1_unifiedpushserver_cr.yaml
```

3. Create a PushApplication CR:

```
kubectl apply -n unifiedpush -f ./deploy/crds/push_v1alpha1_pushapplication_cr.yaml
```

- Ensure the PushApplication shows up in the list of applications in the UPS admin UI
- Ensure the PushApplication has the `pushApplicationId` and `masterSecret` fields populated in the CR status.

4. Create an AndroidVariant CR using the PushApplication ID:

```
cat deploy/crds/push_v1alpha1_androidvariant_cr.yaml | sed -e "s/pushApplicationId: '123456'/pushApplicationId: '$(kubectl get pushapplications -o jsonpath='{.items[0].status.pushApplicationId}')'/" | kubectl apply -f -
```

- Ensure the variant shows up under the Application in the UPS admin UI
- Ensure the AndroidVariant CR has a `status.variantId` populated, and `status.ready` is `true`
- Ensure the AndroidVariant CR has a `finalizer.push.aerogear.org` in `metadata.finalizers`

*Note:* You can do something equivalent to this for the IOSVariant if you've got a cert that you can use

5. Delete the AndroidVariant CR:

```
kubectl delete androidVariant example-androidvariant
```

- Ensure the Variant no longer appears in the UPS admin UI
- Ensure the CR is no longer accessible in Kubernetes

*Note:* You can do something equivalent to this for the IOSVariant if you've created one in step 4.

6. Delete the PushApplication CR:

```
kubectl delete pushApplication example-pushapplication
```

- Ensure the application no longer appears in the UPS admin UI
- Ensure the CR is no longer accessible in Kubernetes

7. Clean up!

```
make cluster/clean
```
